templates/partials/notifications-dropdown.html.twig

```diff
@@ -9,15 +9,17 @@
             </strong>
         </div>
         <div class="col-sm-3 col-sm-offset-5">
-            <a>Mark all as read</a>
+            <a href="{{ path('notifications_mark_read', { return: app.request.pathinfo }) }}">Mark all as read</a>
```

Comment:
Can you implement this as an AJAX request (and then clear the navbar badge when it completes)?

I know that would add a bit more front-end complexity, but it would prevent the page from having to be refreshed. The current behavior might be unexpected for the user. 

If you do end up sticking with this request and return redirect, you'll need to check the return path to make sure it is a relative URL. On the settings page, we added that check, to prevent the chance of a user getting tricked into clicking a link with a malicious return URL.